### ISSUE TITLE
Fitter error on de0_nano

### ISSUE THREAD

**Issue by hansfbaier** [olofk/corescore#15]
`Error (170012): Fitter requires 1420 LABs to implement the design, but the device contains only 1395 LABs
`

**Comment by olofk** [comment]
There is still some work to be done when it comes to predictability. I'm not completely sure which version of Quartus I used for this. Worst case there is even variations when running on different machines. In the meantime, could you locate this line https://github.com/olofk/corescore/blob/master/corescore.core#L285 in your local checkout and see if it works when you change it to 60. Would be good to know how far off we are

**Comment by hansfbaier** [comment]
Hi Olof,
thanks for the answer, I can get 59.
the python tool does not work with some decoding error,
but the output of the serial looks good (all say hello).

Best regards,
Hans

P.S.: Here is the stacktrace:
$  python3 fusesoc_libraries/corescore/sw/corecount.py
Traceback (most recent call last):
  File "fusesoc_libraries/corescore/sw/corecount.py", line 50, in <module>
    curses.wrapper(main)
  File "/usr/lib/python3.8/curses/__init__.py", line 105, in wrapper
    return func(stdscr, *args, **kwds)
  File "fusesoc_libraries/corescore/sw/corecount.py", line 39, in main
    u = umsgpack.unpack(ser)
  File "/usr/local/lib/python3.8/dist-packages/msgpack/__init__.py", line
58, in unpack
    return unpackb(data, **kwargs)
  File "msgpack/_unpacker.pyx", line 211, in msgpack._unpacker.unpackb
msgpack.exceptions.UnpackValueError: Unpack failed: error = 0



Am Fr., 22. Jan. 2021 um 16:38 Uhr schrieb Olof Kindgren <
notifications@github.com>:

**Comment by hansfbaier** [comment]
P.S.: I used: Quartus Version 20.1.0 Build 711 06/05/2020 SJ Lite Edition

Am Sa., 23. Jan. 2021 um 11:53 Uhr schrieb Hans Baier <hansfbaier@gmail.com

**Comment by olofk** [comment]
Thanks for the info. Good to know that it at least was just a small difference. I'll let you know if I figure out which Quartus version I used.

Regarding the msgpack error I just had an idea. Are you using `msgpack` or `umsgpack`? It's very confusing, because there are two packages that implement the same interface, but I have only managed to get `umsgpack` working reliably on my system. If that turns out to be the problem, I should be more clear about this in the instructions and maybe even add a runtime check to detect which version is used
